### ISSUE TITLE
Zoom out [disable attr]

### ISSUE THREAD

**Issue by sgurin** [mapbox/mapbox-gl-js#9172]
Hi, guys.

On the example, https://docs.mapbox.com/mapbox-gl-js/example/navigation/  disable state is stetted on the zoom-in button when you rich the limit https://prnt.sc/qlwb9i but when you zoom out https://prnt.sc/qlwby0 nothing happened with zoom out button. So, how we can disable the zoom out button?

**Comment by ryanhamley** [comment]
Good catch. This is a little tricky. As of https://github.com/mapbox/mapbox-gl-js/pull/9028, the map's default `minZoom` is actually -2, not 0. You can reset that by calling `map.setMinZoom(0)` if you don't need the ability to show your map on small viewports. However, the amount you can zoom out on a map is determined in part by the size of the viewport, even when it's large. Unless the map is displayed at an exact multiple of a 512x512 square, it will be impossible to hit exactly z0 so the zoom out button won't be disabled.

I can think of a couple ways to solve this:

1. we could have a concept of "effective min zoom" which is the zoom that the map is actually able to reach at its current size and disable the zoom out button when that limit is reached

2. we could provide a method to manually disable the zoom buttons so developers can disable them as necessary. something like `navControl.zoomOutButton.disable()`

**Comment by sgurin** [comment]
Hi @ryanhamley 

Thanks a lot for such a detail explanation.

So, there are no methods to disable the zoom out button when your rich min zoom?

From my point zoom out button should have the same behavior as zoom in. It is confusing users when you click on it and nothing happens.

**Comment by ryanhamley** [comment]
>So, there are no methods to disable the zoom out button when your rich min zoom?
>From my point zoom out button should have the same behavior as zoom in. It is confusing users when you click on it and nothing happens.

The zoom out button _does_ behave the same as the zoom in button. The problem is that it's not always (or even usually) possible to actually reach the map's `minZoom`. If you were to set `maxZoom: 22` and `minZoom: 15` for example, you'd see the button's being disabled as expected when you reach the appropriate zoom levels. It's always possible to reach the `maxZoom` since it's not determined by viewport size, but since the `minZoom` is determined in part by viewport size, it's not guaranteed that the map's `minZoom` is actually reachable. That's the source of this particular bug.

**Comment by sgurin** [comment]
Hi, @ryanhamley 

I see. Thanks for clarification.

What about boundaries? Will the bug be the same when we rich the limits of the area?
